projects/output-and-error-handling/client/main.go

```diff
@@ -105,19 +106,20 @@ func (r RealTimeProvider) Until(t time.Time) time.Duration {
 	return time.Until(t)
 }
 
-func convertTime(retryTime string, tp TimeProvider) (int, error) {
-	retrySeconds, err := strconv.Atoi(retryTime)
-	if err == nil && retrySeconds != 0 {
-		return retrySeconds, nil
+func convertTime(tp TimeProvider, retryTime string) (time.Duration, error) {
+	secondsInt, err := strconv.Atoi(retryTime)
+	if err == nil && secondsInt != 0 {
+		retryDuration := time.Duration(secondsInt) * time.Second
+		return retryDuration, nil
 	} else if _, err := time.Parse(http.TimeFormat, retryTime); err == nil {
 		httpTime, err := time.Parse(http.TimeFormat, retryTime)
```

Comment:
Here you can use the output from the first Parse, instead of parsing again
```go
else if val, err := time.Parse(http.TimeFormat, retryTime); err == nil {
		httpTime = val
```
this way, you also don't need to check the error repeatedly